WordPress/Classes/ViewRelated/Jetpack/Install/JetpackInstallPluginHelper.swift

```diff
@@ -1,46 +1,137 @@
+        navigationViewController.modalPresentationStyle = shouldUseFormSheet ? .formSheet : .fullScreen
+        presentingViewController.present(navigationViewController, animated: true) {
+            helper.markOverlayAsShown()
         }
     }
 
-    static func shouldShow(for blog: Blog?) -> Bool {
-        guard let blog, blog.account != nil else {
-            return false
+    init?(_ blog: Blog?, repository: UserPersistentRepository = UserPersistentStoreFactory.instance()) {
+        guard let blog,
+              let siteID = blog.dotComID?.stringValue,
+              blog.account != nil,
+              FeatureFlag.jetpackIndividualPluginSupport.enabled else {
+            return nil
         }
-        let isFeatureEnabled = FeatureFlag.jetpackIndividualPluginSupport.enabled
-        return isFeatureEnabled && blog.jetpackIsConnectedWithoutFullPlugin
+
+        self.blog = blog
+        self.siteIDString = siteID
+        self.repository = repository
     }
 
-    static func shouldShowCard(for blog: Blog?) -> Bool {
-        guard let siteID = blog?.dotComID?.stringValue else {
-            return false
+    func hideCard() {
+        if isCardHidden {
+            return
         }
-        let isCardHidden = cardHiddenSites.contains { $0 == siteID }
-        return shouldShow(for: blog) && !isCardHidden
+        cardHiddenSites += [siteIDString]
     }
 
-    static func hideCard(for blog: Blog?) {
-        guard let blog, let siteID = blog.dotComID?.stringValue else {
+    func markOverlayAsShown() {
+        if isOverlayAlreadyShown {
             return
         }
+        overlayShownSites += [siteIDString]
+    }
+}
+
+// MARK: - Private Helpers
+
+private extension JetpackInstallPluginHelper {
+
+    var isCardHidden: Bool {
+        cardHiddenSites.contains { $0 == siteIDString }
+    }
+
+    var cardHiddenSites: [String] {
+        get {
+            (repository.array(forKey: .cardHiddenSitesKey) as? [String]) ?? [String]()
+        }
+        set {
+            repository.set(newValue, forKey: .cardHiddenSitesKey)
+        }
+    }
 
-        var sites = cardHiddenSites
-        sites.append(siteID)
-        cardHiddenSites = sites
+    var isOverlayAlreadyShown: Bool {
+        overlayShownSites.contains { $0 == siteIDString }
     }
 
-    private struct Constants {
-        static let cardHiddenSitesKey = "jetpack-install-card-hidden-sites"
+    var overlayShownSites: [String] {
+        get {
+            (repository.array(forKey: .overlayShownSitesKey) as? [String]) ?? [String]()
+        }
+        set {
+            repository.set(newValue, forKey: .overlayShownSitesKey)
+        }
     }
 
+    var shouldPromptInstall: Bool {
+        blog.jetpackIsConnectedWithoutFullPlugin
+    }
+}
+
+// MARK: - String Constants
+
+private extension String {
+    static let cardHiddenSitesKey = "jetpack-install-card-hidden-sites"
+    static let overlayShownSitesKey = "jetpack-install-overlay-shown-sites"
 }
```

Comment:
I've always liked having constants this way over putting them in a `struct`. But when I originally tried that I was told to change it 😅. Not saying you have to change it here but it would be cool if we could move towards this type of constants.

Reply:
Gotcha. Yeah, I don't think we have strict guidelines on defining constants. And while doing this way is more concise, we'd have to create the `struct` or `enum` anyways when a non-string constant comes along.

Come to think of it, in the codebase it looks like I'm the only one who creates string constants this way 😂. I'll think about it and may push a change later.